server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -248,20 +248,21 @@ protected static Collection<String> resolveExpressions(Context context, String..
             if (expressions == null || expressions.length == 0 || expressions.length == 1 && Metadata.ALL.equals(expressions[0])) {
                 return List.of();
             } else {
-                return ExplicitResourceNameFilter.filterUnavailable(
-                    context,
-                    DateMathExpressionResolver.resolve(context, List.of(expressions))
-                );
+                // Do not evaluate remote expressions on local node, instead let remote node evaluate them when request arrives there
+                List<String> localExpressions = RemoteResourceNameFilter.filterRemoteIndexNames(List.of(expressions));
+                return ExplicitResourceNameFilter.filterUnavailable(context, DateMathExpressionResolver.resolve(context, localExpressions));
```

Comment:
I think the change to not "resolve" remote index names should be done in a caller to this method, see for example `TransportSearchAction#resolveLocalIndices`. This class is already very complex and brittle and I would suggest we avoid having it also deal with remote names, if possible (eg what if remote names are wildcards?).